# Cálculo - Tarefa 02

1- Calcule de forma numérica a integral da função $x^3$ avaliada entre os pontos $1$ e $2$. Identifique o número de retângulos que você usou.  
2- Calcule essa área de forma analítica.

1- Calcule de forma numérica a integral da função $x^3$ avaliada entre os pontos $1$ e $2$. Identifique o número de retângulos que você usou.  

In [1]:
import numpy as np

In [2]:
f = lambda x: x**3

def calcula_area(a, b, func, quantidade_retangulo):
    delta = (b-a)/quantidade_retangulo 
    x_cols = np.linspace(a, b-delta, quantidade_retangulo)+delta/2 
    y_cols = f(x_cols)
    return sum(delta*y_cols)

In [3]:
calcula_area(1, 2, f, 50)

3.749850000000001

2- Calcule essa área de forma analítica.

$$
\int_{a}^{b} f(x)dx = F(b) - F(a)
$$

$$
f(x) = x^3 \rightarrow F(x) = \frac{1}{4} x^4 + C,
$$

$$
\int_{1}^{2} x^3 dx = F(2) - F(1) = \left( \frac{1}{4} \times 2^4 \right) - \left( \frac{1}{4} \times 1^4 \right) = \frac{16}{4} - \frac{1}{4} = \frac{15}{4} = \mathbf{3.75}
$$
